File: tests/source_samples/javascript/control_flow/test_control_flow.js
// JavaScript Control Flow Test Suite

// if/else if/else
let num = 0;
if (num > 0) {
    // positive
} else if (num < 0) {
    // negative
} else {
    // zero
}

// Nested if
let x = 5, y = 10;
if (x > 0) {
    if (y > 0) {
        // both positive
    }
}

// switch statement
let day = 'Monday';
switch (day) {
    case 'Monday':
        break;
    case 'Tuesday':
        break;
    default:
        break;
}

// Switch with fall-through
const value = 2;
switch (value) {
    case 1:
    case 2:
    case 3:
        // handles 1, 2, or 3
        break;
    default:
        // other values
}

// for loop
for (let i = 0; i < 3; i++) {
    // loop body
}

// for...in loop (objects)
for (const key in {a:1, b:2}) {
    // iterate over object keys
}

// for...of loop (iterables)
for (const val of [1, 2, 3]) {
    // iterate over array values
}

// while loop
let k = 3;
while (k > 0) {
    k--;
}

// do...while loop
let count = 0;
do {
    count++;
} while (count < 3);

// break and continue
for (let i = 0; i < 10; i++) {
    if (i === 2) continue;
    if (i === 5) break;
}

// Labeled statements
outer: for (let i = 0; i < 3; i++) {
    for (let j = 0; j < 3; j++) {
        if (i === 1 && j === 1) {
            break outer;
        }
    }
}

// try/catch/finally
try {
    throw new Error("Intentional error");
} catch (e) {
    // handle error
} finally {
    // always executes
}

// Multiple catch (proposed feature, using if/else pattern)
try {
    // risky code
} catch (e) {
    if (e instanceof TypeError) {
        // handle TypeError
    } else if (e instanceof ReferenceError) {
        // handle ReferenceError
    } else {
        // handle other errors
    }
}

// Conditional (ternary) operator
const result = num > 0 ? "positive" : "non-positive";

// Nullish coalescing
const value1 = null ?? "default";
const value2 = undefined ?? "fallback";

// Optional chaining
const obj = { a: { b: { c: 1 } } };
const deepValue = obj?.a?.b?.c;
const missingValue = obj?.x?.y?.z;

console.log("Control flow test completed");

File: tests/source_samples/python/control_flow/test_control_flow.py
# Python Control Flow Test Suite

# if/elif/else statements
x = 10
if x > 5:
    result = "greater"
elif x == 5:
    result = "equal"
else:
    result = "less"

# Nested if statements
x, y = 5, 10
if x > 0:
    if y > 0:
        result = "both positive"
    else:
        result = "x positive, y not"
else:
    result = "x not positive"

# for loop
total = 0
for i in range(5):
    total += i

# for/else
for i in [1, 2, 3]:
    if i == 4:
        break
else:
    pass  # Loop completed without break

# Nested for loops
for i in range(3):
    for j in range(3):
        pass

# while loop
count = 5
while count > 0:
    count -= 1

# while/else
count = 3
while count > 0:
    count -= 1
    if count == -1:  # This will not happen
        break
else:
    pass  # Loop completed without break

# break and continue
found_even = False
for i in range(10):
    if i % 2 != 0:
        continue
    if i > 5:
        break
    if i == 4:
        found_even = True

# try/except/finally
try:
    1 / 0
except ZeroDivisionError:
    pass  # Handle division by zero
finally:
    pass  # Always executes

# Multiple except clauses
try:
    risky_operation = 1 / 0
except ZeroDivisionError:
    error_handled = True
except Exception as e:
    other_error = str(e)
finally:
    cleanup_done = True


# with statement (Context Managers)
class MyContext:
    def __enter__(self):
        return "Hello from with"

    def __exit__(self, exc_type, exc_val, exc_tb):
        return False


with MyContext() as cm:
    pass

# Multiple context managers
with MyContext() as ctx1, MyContext() as ctx2:
    multi_context_result = ctx1 + ctx2

# Assert statement
assert 1 == 1, "This should never fail"


# Pass statement
def empty_func():
    pass


print("Control flow test completed")
